components/profile/my-network-settings/registry-settings.tsx

```diff
@@ -13,29 +13,88 @@ import { WarningSpan } from "components/warning-span";
 import { useAppState } from "contexts/app-state";
 import { useNetworkSettings } from "contexts/network-settings";
 
+import { ParameterValidator, REGISTRY_LIMITS } from "helpers/registry";
+
+import { RegistryParameters } from "types/dappkit";
+
 import useBepro from "x-hooks/use-bepro";
 import { useNetwork } from "x-hooks/use-network";
 
+type Executing = "bountyFees" | "creationFee" | "creationAmount";
+
 export default function RegistrySettings({ isGovernorRegistry = false }) {
   const { t } = useTranslation(["common", "custom-network", "setup"]);
-  const [networkCreationFeePercentage, setNetworkCreationFeePercentage] = useState<string>()
-  const [lockAmountForNetworkCreation, setLockAmountForNetworkCreation] = useState<string>()
-  const { fields, settings } = useNetworkSettings();
+
+  const [executingTx, setExecutingTx] = useState<Executing>();
+  const [networkCreationFeePercentage, setNetworkCreationFeePercentage] = useState<string>();
+  const [lockAmountForNetworkCreation, setLockAmountForNetworkCreation] = useState<string>();
+  
   const {state} = useAppState();
   const { updateActiveNetwork } = useNetwork();
+  const { fields, settings } = useNetworkSettings();
   const { handleFeeSettings, handleAmountNetworkCreation, handleFeeNetworkCreation } = useBepro();
 
+  function validateLimits(param: RegistryParameters, value: string | number) {
+    if (ParameterValidator(param, value)) return undefined;
+
+    const { min, max } = REGISTRY_LIMITS[param] || {};
+
+    if (min !== undefined && max !== undefined)
+      return t("setup:registry.errors.exceeds-limit", { min, max });
+    else
+      return t("setup:registry.errors.greater-than", { min });
```

Comment:
if min is undefined, will it show "needs to be greater than 0"?

Reply:
No, but I just sent a commit refactoring this function